### PR TITLE
Use lowercase sensitive "github.com/sirupsen/logrus"

### LINKED ISSUE #1261: Use lowercase sensitive "github.com/sirupsen/logrus"
Our vendor dependency tree references "github.com/Sirupsen/logrus" which was changed to "github.com/sirupsen/logrus" in https://github.com/sirupsen/logrus/pull/384. This causes an issue with updating other vendor packages which are dependent on logrus. We should find a way to clean this up in order to update certain packages to more recent versions (ex. github.com/atlassian/gostatsd).

```
Solving failure: No versions of github.com/atlassian/gostatsd met constraints:
	5.4.0: Could not introduce github.com/atlassian/gostatsd@5.4.0 due to a case-only variation: it depends on "github.com/sirupsen/logrus", but "github.com/Sirupsen/logrus" was already established as the case variant for that project root by the following other dependers:
	(root)
	github.com/docker/docker@v1.13.1
	github.com/tmc/grpc-websocket-proxy@timestamp_stdtime_example
```

### PR DESCRIPTION
**Pull request by nikkiattea**
## What is this change?

Updates dependency "Sirupsen/logrus" to "sirupsen/logrus." Updates other applicable dependencies to versions which use the latter. 

## Why is this change necessary?

Closes #1261.

## Does your change need a Changelog entry?

Yas.

## Do you need clarification on anything?

Nah.

## Were there any complications while making this change?

~~Updating gostatsd to 5.4.1 removes a deprecated function `NewServer`. I had to rewrite that function for our purposes.~~

### CODE REVIEW

**Comment by cyphus** [comment]
Changes look good from here, going to make sure I can build locally before giving the official thumbs up.

File: agent/statsd_server.go [hunk 1]
```diff
@@ -31,6 +32,38 @@ func NewStatsdServer(a *Agent) *statsd.Server {
 	return s
 }
 
+// NewServer will create a new statsd Server with the default configuration.
+func NewServer() *statsd.Server {
```
Comment (echlebek) [inline agent/statsd_server.go]: This change seems unrelated to the PR topic, a mistake?

Comment (nikkiattea) [reply agent/statsd_server.go]: It seems so. NewServer is deprecated in gostatsd version 5.4.1 and we had to update gostatsd for sirupsen/logrus compatibility. I just used the most recent release as that’s what we originally attempted, which prompted this logrus issue.

Comment (nikkiattea) [reply agent/statsd_server.go]: Yes! For some reason I assumed the function was deprecated _before_ the logrus fix. But it wasn't, so we can totally roll this back to 2.4.0.
